### PR TITLE
IMP: Improved error message when inappropriate primitive passed

### LINKED ISSUE #222: BUG: Opaque error when inappropriate value passed to primitive parameter
**Bug Description**
Changes to q2cli in 2019.4 (likely [_convert_primitive](https://github.com/qiime2/q2cli/blame/master/q2cli/click/type.py)/[parse_primitive](https://github.com/qiime2/qiime2/blame/master/qiime2/core/type/util.py) are causing opaque `_COERCE_ERROR`s when users pass inappropriate values to primitive parameters. 

**Steps to reproduce the behavior**
1. Activate a `2019.4` or `2019.7.dev` Conda environment
2. Pass an inappropriate value to a command with a `--p-` parameter.
3. See terrifying error message

**Expected behavior**
A more understandable error message should be rendered. 
Falling back on the [`check_types`](https://github.com/qiime2/qiime2/blob/3e053cba101ace49ba52e16cba0bd6966973b0b3/qiime2/core/type/signature.py#L323) error messages might be nice.

**Computation Environment**
- OS: linuxMint 18.3, OpenSUSE Tumbleweed
- QIIME 2 Releases 2018.11-2019.7.dev

**Comments**
1. Impacts only `--p-` arguments
2. Does not impact Artifact API

**References**
1. Forum [xref](https://forum.qiime2.org/t/error-in-run-dada2/10523)

### PR DESCRIPTION
**Pull request by Oddant1**
Closes #222 by making the error message look more in line with the current error that can be thrown by `type_cast_value`. This is the error message:

Selection_095

### CODE REVIEW

File: q2cli/click/type.py [hunk 1]
```diff
@@ -192,7 +192,15 @@ def _convert_metadata(self, value, param, ctx):
...
         import qiime2.sdk.util
 
-        return qiime2.sdk.util.parse_primitive(self.type_expr, value)
+        try:
+            return qiime2.sdk.util.parse_primitive(self.type_expr, value)
+        except ValueError:
+            args = ', '.join(map(repr, value))
+            expr = qiime2.sdk.util.type_from_ast(self.type_ast)
+            raise click.BadParameter(
+                'recieved <%s> as an argument, which is incompatible'
```
Comment (thermokarst) [inline q2cli/click/type.py]: Drive-by comment: 
```suggestion
                'received <%s> as an argument, which is incompatible'
```

Comment (Oddant1) [reply q2cli/click/type.py]: It's misspelled all over the place and I copy-pasted :laughing: I'll fix them all

**Comment by thermokarst** [comment]
Hey there @ChrisKeefe, can you please review this PR when you get a chance? Please address any changes you might want made to @ebolyen & I - we are teaming up to run this one across the finish line!
